### DISCUSSION TITLE
Dealing with errors in `5.0.0`

### DISCUSSION THREAD

**Discussion by mehroozkhan** [sbertix/Swiftagram#183] [Q&A] [answered]
Hi, 

i am using your library in storyboard project (not using swiftUI) , how can I call the api to get user , I saw your example : 

```
Endpoint.user(secret.identifier)
            .unlock(with: secret)
            .session(.instagram)
            .sink { <#Subscribers.Completion<Error>#> in
                <#code#>
            } receiveValue: { User.Unit in
                <#code#>
            }
```

receiveValue closure is not optional , how can I check if api failed to get user for identifier ?

**Comment by mehroozkhan** [comment]
Never mind I figured it out. 
Is there any workaround if I don't want to call `.store(in: &bin)` at the end, how can i avoid warning?

**Reply by sbertix** [reply]
> Never mind I figured it out. 
> 
> Is there any workaround if I don't want to call `.store(in: &bin)` at the end, how can i avoid warning?

You need to retain your call somehow, using a `Set<AnyCancellable>` is the easiest/most compact way. 

Keep in mind that if you don't retain your sinked `Publisher` the request is immediately cancelled. 

If you need to find out more about **Combine** `Publisher`s, sinks, `Cancellable`s, etc. there's a lot of good content online, as this is not immediately related to **Swiftagram**.

**Comment by sbertix** [comment, marked as answer]
I'm still gonna add a reply for future reference, in case anybody else wonders. 

```swift 
// Make sure this is retained!
var bin: Set<AnyCancellable> = []

Endpoint.user(secret.identifier)
            .unlock(with: secret)
            .session(.instagram)
            .sink { 
               switch $0 {
               case .failure(let error): break
               default: break
               }
            } receiveValue: { 
               print("output", $0)
            }
            .store(in: &bin)
```

**Reply by mehroozkhan** [reply]
Got it, thank you, 

btw there is an issue in `tag` endpoint: 

![Screen Shot 2021-05-07 at 1 40 26 AM](https://user-images.githubusercontent.com/17316374/117362900-4727f480-aed5-11eb-8376-547da5c12039.png)

offset for pagination is working fine on other endpoints but not for the above endpoint.

**Reply by sbertix** [reply]
It's not an issue. 
Tag offsets are not optional `String`s, but `Section.Offset`s, also including the page and current media identifiers. 

It's not a problem as you should only ever populate the parameter with the `offset` you get from the actual response, not random string laying around, e.g. ask for the first page, cache the `offset` and then pass it when you need to fetch the second one. 

Check out `Section.Offset` for the definition.

**Reply by mehroozkhan** [reply]
it still gives me error, even if I pass non optional value, no error in other apis even if I pass optional value : 

![Screen Shot 2021-05-07 at 2 19 08 AM](https://user-images.githubusercontent.com/17316374/117366962-f0bdb480-aeda-11eb-997f-e0e8f8cf04df.png)

**Reply by sbertix** [reply]
Cause you're still passing the same `String`, instead of the `Section.Offset` you should use, like I told you in my previous comment 😊

Please, this is now extremely off topic and I've already told you what's wrong with your code... just apply what I told you in my comment before.
